File: array/remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int n = nums.length;
        int count = 1;
        int i = 1;

        while (i < n) {
            if (nums[i] != nums[i - 1]) {
                nums[count] = nums[i];
                count++;
            }
            i++;
        }

        return count;    
    }
}

File: array/remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        n = len(nums)
        prev = nums[0]
        i = 1
        j = 1
        while i < n:
            if (nums[i] != prev):
                nums[j] = nums[i]
                prev = nums[j]
                j += 1
                i += 1
            else:
                i += 1
        return j
            
                
        